boot-qemu.sh

```diff
@@ -177,24 +184,40 @@ function setup_qemu_args() {
     [[ -f ${KERNEL} ]] || die "${KERNEL} does not exist!"
 }
 
-
 # Invoke QEMU
 function invoke_qemu() {
     ${INTERACTIVE} || QEMU=( timeout "${TIMEOUT:=3m}" unbuffer "${QEMU[@]}" )
     if ${GDB:=false}; then
-        # Print message in bold green
-        printf '\033[01;32m'
-        echo
-        echo "Starting QEMU with GDB connection on port 1234..."
-        echo
-        echo "Use:"
-        echo
-        printf '\ttarget remote :1234\n'
-        echo
-        echo "to connect"
-        echo
-        printf '\033[0m'
-        QEMU=( "${QEMU[@]}" -s -S )
+        while true; do
+            if lsof -i:1234 &>/dev/null; then
+                red "Port :1234 already bound to. QEMU already running?"
+                exit 1
+            fi
+            green "Starting QEMU with GDB connection on port 1234..."
+            # Note: no -serial mon:stdio
+            "${QEMU[@]}" \
+              "${QEMU_ARCH_ARGS[@]}" \
+              -display none \
+              -initrd "${ROOTFS}" \
+              -kernel "${KERNEL}" \
+              -m "${QEMU_RAM:=512m}" \
+              -nodefaults \
+              -s -S &
+            QEMU_PID=$!
+            green "Starting GDB..."
+            gdb "${KBUILD_DIR}/vmlinux" -ex "target remote :1234"
+            red "Killing QEMU..."
+            kill -9 "${QEMU_PID}"
+            wait "${QEMU_PID}" 2>/dev/null
+            while true; do
+              read -p "Rerun [Y/n/?] " yn
+              case $yn in
+                [Yy]* ) break ;;
+                [Nn]* ) exit 0 ;;
+                * ) break ;;
```

Comment:
We should probably start using some sort of shell formatter like we use [YAPF](https://github.com/google/yapf) for the Python code in [tc-build](https://github.com/ClangBuiltLinux/tc-build) since I use 4 spaces and you use 2 and I frequently have to adjust my thinking in the CI repo. Should just make it brainless.

Not going request a change, just merely a meta comment.

@msfjarvis: Don't you use a formatter for your scripts?

Reply:
yeah I use [shfmt](https://github.com/mvdan/sh)

Reply:
```yaml
on: [push, pull_request]
name: Check script formatting
jobs:
  shfmt-check:
    name: shfmt
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - uses: actions/cache@v1
      with:
        path: ~/go/pkg/mod
        key: ${{ runner.os }}-go
        restore-keys: |
          ${{ runner.os }}-go-
    - name: Install shfmt
      run: GO111MODULE=on go get mvdan.cc/sh/v3/cmd/shfmt
    - name: Run shellcheck.
      run: PATH=$HOME/go/bin:$PATH shfmt -d -i 2 ${SCRIPTS_TO_TEST}
```

this should be enough to error out in case of violations and get everything to 2 space indent.

Reply:
eh, sorry, I'll reformat to 4 spaces